File: typescript/darts/darts.ts
export function score(x: number, y: number): number {
  const r = Math.sqrt(x*x + y*y);

  if (r <= 1) {
    return 10;
  }

  if (r <= 5) {
    return 5;
  }

  if (r <= 10) {
    return 1;
  }

  return 0;
}

File: c/darts/darts.c
#include "darts.h"
#include <math.h>

int score(struct coordinate_t landing_position) {
   float r = sqrt(pow(landing_position.x, 2) + pow(landing_position.y, 2)); 
   if (r > 10) {
      return 0;
   }
   if (r > 5) {
      return 1;
   }
   if (r > 1) {
      return 5;
   }
   return 10;
}
